ts/features/bonus/bpd/store/reducers/details/activation/index.ts

```diff
@@ -117,4 +120,18 @@ export const bpdUnsubscriptionSelector = createSelector(
   unsubscription => unsubscription
 );
 
+/**
+ * Return the prefill text based on the iban upsert or iban inserted by user
+ * if the user tried to add a new iban (upsertIban.value !== undefined) return that iban
+ * else try to return the actual iban getValue(iban)
+ *
+ */
+export const bpdIbanPrefillSelector = createSelector(
```

Comment:
It's hard to understand what this selector returns.
Btw I guess it never returns undefined. If it is right, could this be a refactoring?

```typescript
export const bpdIbanPrefillSelector = createSelector<
  GlobalState,
  RemoteValue<string | undefined, Error>,
  UpsertIBAN,
  string
>([bpdIbanSelector, bpdUpsertIbanSelector], (iban, upsertIban): string =>
  fromNullable(upsertIban.value as string).getOrElse(
    fromNullable(getValue(iban)).getOrElse("")
  )
);
```

Reply:
Changed the signature in 3b50508 in order to avoid the "undefined"